### PR TITLE
Use logical string comparison for vehicle list order

### PR DESCRIPTION
**Pull request by AaronVanGeffen**
Before:
Screenshot (4)

After:
Screenshot (2)

### CODE REVIEW

File: src/OpenLoco/Utility/String.cpp [hunk 1]
```diff
@@ -12,6 +12,40 @@
 
 namespace OpenLoco::Utility
 {
+    // Case-insensitive logical string comparison function
+    int32_t strlogicalcmp(const char* s1, const char* s2)
```
Comment (IntelOrca) [inline src/OpenLoco/Utility/String.cpp:16]: Change to use `std::string_view`?

Comment (AaronVanGeffen) [reply src/OpenLoco/Utility/String.cpp:16]: Hmm. Normally, I'd be for that, but the entire function body is working with C char pointers. I would rather not have to rewrite the function, considering it leans pretty heavily on C string library functions (`isdigit`, `toupper`, `strtoul`).

Comment (duncanspumpkin) [reply src/OpenLoco/Utility/String.cpp:16]: You would have to use `from_chars` instead of strtoul but i think the other functions would work fine.

Comment (IntelOrca) [reply src/OpenLoco/Utility/String.cpp:16]: > I would rather not have to rewrite the function

Why not?

Comment (duncanspumpkin) [reply src/OpenLoco/Utility/String.cpp:16]: https://godbolt.org/z/YsKhhGPrT this is my attempt at recreating it with string_views

Comment (duncanspumpkin) [reply src/OpenLoco/Utility/String.cpp:16]: Think you can even make it `constexpr` but no real need.

Comment (AaronVanGeffen) [reply src/OpenLoco/Utility/String.cpp:16]: @duncanspumpkin Thanks, please feel free to push it as a commit. :)

Comment (duncanspumpkin) [reply src/OpenLoco/Utility/String.cpp]: done

Comment (duncanspumpkin) [reply src/OpenLoco/Utility/String.cpp]: Note that this isn't utf8 friendly but I don't think that is really an issue atm but might be a consideration if we are porting this up to OpenRCT2.

Comment (duncanspumpkin) [reply src/OpenLoco/Utility/String.cpp]: Urgh how annoying its not supported by the compiler versions we are using. We will need to revert my commit and stick to the original code.

Comment (IntelOrca) [reply src/OpenLoco/Utility/String.cpp]: @duncanspumpkin here you go:

https://godbolt.org/z/8cqvWefKf

Comment (duncanspumpkin) [reply src/OpenLoco/Utility/String.cpp]: okay updated it with that snippet

**Review by AaronVanGeffen — COMMENTED** [review]
Thanks. The changes from Duncan and Ted look good to me.

**Pull request merged**